libcudacxx/include/cuda/std/__iterator/reverse_iterator.h

```diff
@@ -278,10 +278,7 @@ class _LIBCUDACXX_TEMPLATE_VIS reverse_iterator
   {
     auto __xtmp = __x.base();
     auto __ytmp = __y.base();
-    _CUDA_VRANGES::iter_swap(--__xtmp, --__ytmp);
-#    if defined(_CCCL_COMPILER_ICC)
-    _LIBCUDACXX_UNREACHABLE();
-#    endif // _CCCL_COMPILER_ICC
```

Comment:
I believe that was an actual bug that was never caught because we do not run ICC in CI